v2/tools/asoctl/internal/crd/cleaner.go

```diff
@@ -199,3 +200,9 @@ func removeMatchingStoredVersions(oldVersions []string, versionRegexp *regexp.Re
 
 	return newStoredVersions, matchedStoredVersion
 }
+
+// getVersionFromStoredVersion returns the public (non-storage) API version for a given version
+func getVersionFromStoredVersion(version string) string {
```

Comment:
So technically this fix is enough and we don't need the fix in `group_version_kind_aware`, but I put both just to be on the safe side.

Without this change we see the following error:
> E0406 00:01:43.131605       1 generic_reconciler.go:348] controllers/VirtualNetworksSubnetController "msg"="Encountered error impacting Ready condition" "error"="Reason: Failed, Severity: Warning, RetryClassification: RetrySlow, Cause: unable to get spec from network.azure.com/v1api20201101storage, Kind=VirtualNetworksSubnet: getting versioned spec: unable to create new network.azure.com/v1alpha1api20201101, Kind=VirtualNetworksSubnet: no kind \"VirtualNetworksSubnet\" is registered for version \"network.azure.com/v1alpha1api20201101\" in scheme \"pkg/runtime/scheme.go:100\"" "name"="samplesubnet" "namespace"="default"

This is because the `asoctl crd clean` command was using the storage version to do the update not the API version, which wasn't correctly changing the latest used API version.